kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -794,133 +683,98 @@ define([
-                        var link_id = new UUID(4).format();
-                        $ul.append(
-                            $.jqElem('li').append(
-                                $.jqElem('a')
-                                    .attr('id', link_id)
-                                    .attr('href', '#')
-                                    .append(v.name || v.URL)
-                                    .prop('download', true)
-                                    .attr('download', 'download')
-                            )
-                        );
-
-                        setTimeout(() => {
-                            $('#' + link_id).on('click', (e) => {
-                                e.preventDefault();
-                                e.stopPropagation();
-                                $('#' + iframe_id).attr(
-                                    'src',
-                                    self.importExportLink(v.URL, v.name || 'download-' + i)
-                                );
-                            });
-                        }, 1);
-                    });
-
-                    const $iframe = $.jqElem('iframe').attr('id', iframe_id).css('display', 'none');
-
-                    var body = $.jqElem('div').append($ul).append($iframe).html();
-
-                    ui.setContent(
-                        'downloadable-files',
-                        ui.buildCollapsiblePanel({
-                            title: 'Files',
-                            name: 'downloadable-files-toggle',
-                            hidden: false,
-                            type: 'default',
-                            classes: ['kb-panel-container'],
-                            body: body,
-                        })
-                    );
-                }
+            if (this.options.showFiles && report.file_links && report.file_links.length) {
+                this.$mainPanel.append(div({ dataElement: 'downloadable-files' }));
+                const fileLinksPanelBody = this.buildFileLinksPanel(report.file_links, ui, events);
+                ui.setContent(
+                    'downloadable-files',
+                    ui.buildCollapsiblePanel({
+                        title: 'Files',
+                        name: 'downloadable-files-toggle',
+                        hidden: false,
+                        type: 'default',
+                        classes: ['kb-panel-container'],
+                        body: fileLinksPanelBody,
+                    })
+                );
             }
-
             events.attachEvents();
-
             this.loading(false);
+            return Promise.all(renderPromises);
         },
         loading: function (isLoading) {
             if (isLoading) {
-                this.showMessage('<i class="fa fa-spinner fa-spin"></i>');
+                const span = html.tag('span'),
+                    i = html.tag('i'),
+                    loadingMsg = span(
```

Comment:
didn't you create a loading spinner function somewhere? Ah yes, `UI.loading`. Given how short this is, though, it's not worth changing it.

Reply:
I completely forgot about that! Again, consistency helps. Updated.